test-e2e/test_zookeeper_backup.py

```diff
@@ -0,0 +1,407 @@
+"""
+Tests for the ZooKeeper backup/restore scripts.
+"""
+
+import logging
+import subprocess
+import textwrap
+import uuid
+from pathlib import Path
+from typing import List
+
+import pytest
+from _pytest.fixtures import SubRequest
+from cluster_helpers import wait_for_dcos_oss
+from dcos_e2e.backends import Docker
+from dcos_e2e.cluster import Cluster
+from dcos_e2e.node import Output
+from kazoo.client import KazooClient
+from kazoo.exceptions import NoNodeError
+
+
+LOGGER = logging.getLogger(__name__)
+
+# Arbitrary value written to ZooKeeper.
+FLAG = b'flag'
+
+
+def _zk_set_flag(hosts: str, ephemeral: bool = False) -> str:
+    """
+    Store the `FLAG` value in ZooKeeper in a random Znode.
+    """
+    znode = '/{}'.format(uuid.uuid4())
+    zk_client = KazooClient(hosts=hosts)
+    zk_client.start()
+    zk_client.create(znode, makepath=True, ephemeral=ephemeral)
+    zk_client.set(znode, FLAG)
+    zk_client.stop()
+    return znode
+
+
+def _zk_check_flag(hosts: str, znode: str) -> None:
```

Comment:
A key part of this is that it raises `NoNodeError` if the flag is not there.
Instead, what do you think of:

```
def _zk_flag_exists(hosts: str, znode: str, flag: bytes) -> bool:
    ...
```

Then, a test doesn't need the `with` but instead `assert not _zk_flag_exists(...)` and `assert _zk_flag_exists(...)`.